### ISSUE TITLE
ChangeHost "looping" (sorta)

### ISSUE THREAD

**Issue by NiklasBeierl** [PixlOne/logiops#195]
# Setup:
Logid v0.2 from aur logiops
Wireless Mouse MX Master 2S connect to my desktop via Unify USB (Channel 1) And to my laptop via bluetooth (Channel 2).
My config:  `/etc/logid.cfg`
```
devices: ({
name: "Wireless Mouse MX Master 2S";
buttons: (
{ cid: 0xc3;
  action: { type: "ChangeHost"; host: "next"; }
});
});
```
## What I want to achieve:
Using the "thumb button" to cycle through the hosts. / Switch between 1 and 2 at least. I don't wanna have to flip my mouse over to achieve that. :) 
## Observed behavior:
I launched logid with the config above, pressed the thumb button and it switches to my laptop, so far so good. Output:
```
R ~/aur/logiops (master) sudo logid  
[WARN] Error adding device /dev/hidraw1: std::exception
[INFO] Detected receiver at /dev/hidraw0
[INFO] Device Keyboard K380 not configured, using default config.
[INFO] Device found: Keyboard K380 on /dev/hidraw2:255
[INFO] Device found: Wireless Mouse MX Master 2S on /dev/hidraw0:1
[INFO] /dev/hidraw0:1 fell asleep.
```
First problem: I cant' switch to the next (3) host, once connected to the laptop. I suspect I would have to run logid on my laptop as well? I could set it up to go to the "prev" host there, such that I can toggle back to the desktop.

BUT: There is another problem: 
I used the button at the bottom of the mouse to cycle to 3, which worked.
Then I pressed again to go to "1": The led for "1" briefly lights up and then immediately jumps to "2". logid produces the following output:
```
[INFO] /dev/hidraw0:1 woke up.
[INFO] /dev/hidraw0:1 woke up.
[ERROR] Caught HID++ 1.0 error while trying to initialize /dev/hidraw0:1: Resource error
[WARN] Error ignored on detached thread/task: Resource error
[INFO] /dev/hidraw0:1 fell asleep.
```
From here on I am stuck in a loop, whenever I cycle to "1" it jumps to "2" and logid produces the above output. :/ 
The only way to connect to 1 again is: Stopping logid, turning the mouse off for a few seconds, turning it back on and selecting 1 via the button at the bottom.

**Comment by Cherkah** [comment]
hello
what you are expecting from logiops is the flow support!
but unfortunately this feature is not implemented to its code cf #23 
like you i was expecting to find this feature by installing it but neither logiops nor solaar provides 'flow'

this missing should be indicate in the wiki for everyone who expecting it by installing logiops... :/

**Comment by barolo** [comment]
@NiklasBeierl while it's not perfect, one of the solutions with solar is to script it, one script from A to B and one from B to A, you can launch it in multiple ways, assign shortcut, click it in the dock, active screen corner and so on.  I'm using it that way to switch both of my Logitech devices at once, and it's instant.

**Comment by NiklasBeierl** [comment]
@Cherkah not really. Flow switches the devices as soon as I "leave the screen" on one of them, along with shared clipboard. If I wanted that, I would use something like [synergy](https://symless.com/synergy). I just want the thumb button on my mouse to act exactly like the "switch hosts" button on the bottom side of the device.

**Comment by Cherkah** [comment]
hello,
i come back to report trouble with logiops about changeHost with the thum button.

2 PC: one with unifying gongle & one with bluetooth.
logiops installed on both
logiops.cfg:
```
devices: ({
2   │ name: "Wireless Mouse MX Master 3";
3   │ buttons: (
4   │ { cid: 0xc3;
5   │   action: { type: "ChangeHost"; host: "previous"; }
6   │ });
7   │ });
```

1) logiops must be restart after boot #204 to works even if systemd service enable
2) i can switch between device only once. after this the soft do not works any more unless i restart the service
below the error issue:
`[WARN] Error adding device /dev/hidraw1: std::exception`
